source/features/bypass-checks.tsx

```diff
@@ -1,19 +1,20 @@
+import mem from 'mem';
 import onetime from 'onetime';
 import {observe} from 'selector-observer';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
-import fetchDom from '../helpers/fetch-dom';
+import * as api from '../github-helpers/api';
+import {getRepoURL} from '../github-helpers';
 
-async function bypass(detailsLink: HTMLAnchorElement): Promise<void> {
-	const directLink = await fetchDom<HTMLAnchorElement>(
-		detailsLink.href,
-		'[data-hydro-click*="check_suite.external_click"]'
-	);
+const getDirectLink = mem(async (runNumber: number): Promise<string> => {
+	const directLink = await api.v3(`repos/${getRepoURL()}/check-runs/${runNumber}`);
+	return directLink.details_url;
+});
 
-	if (directLink) {
-		detailsLink.href = directLink.href;
-	}
+async function bypass(detailsLink: HTMLAnchorElement): Promise<void> {
+	const runNumber = detailsLink.href.split(/\/|=/).pop();
```

Comment:
```suggestion
	const runNumber = detailsLink.href.split(/[/=]/).pop();
```

Reply:
I'm only seeing links with `/` however, where did you see the `=`?

Reply:
Hovercards

Reply:
I don't see it there, but in the PR Mergeability box the URL is toward a PR page:

```
https://github.com/sindresorhus/refined-github/pull/3629/checks?check_run_id=1223857819
```

Reply:
Let's make this explicit:

```suggestion
	const runId = new URLSearchParams(detailsLink).get('check_run_id') ?? detailsLink.pathname.split('/').pop();
```

Reply:
Sorry I mixed up what was what. Explicit is much better!